### ISSUE TITLE
compile groovy dsl is not converted  to kotlin

### ISSUE THREAD

**Issue by mystilleef** [bernaferrari/GradleKotlinConverter#4]
A lot of older programs, especially non-Android projects, still use the "**compile**" dependency directive instead of "**implementation**". Unfortunately, the converter doesn't convert the "**compile**" directive from Groovy functions to Kotlin. Not sure if you're aware of this yet.

**Comment by bernaferrari** [comment]
I ended up not including that conversion because those shouldn't even build on newer gradle versions. The idea of the plugin was Gradle Groovy -> Gradle Kotlin, not Gradle 4 to Gradle 5 and Gradle 5 to Gradle 5 Kotlin. But, it is not hard to include that transformation. Do you really want it?

**Comment by mystilleef** [comment]
As far as I know, **compile** is deprecated only on Android. On vanilla non-Android projects **compile** is discouraged, but it's still in wide use. 

I personally would like a feature to change **compile** to **implementation** but that might be outside the scope of this project. 

Most of the projects that I have to update still use **compile**, even  Android projects. 

So yes, support for **compile** would make life easier for developers who have to update and maintain older build configurations that haven't been updated in years, or that have been sloppily written.

**Comment by bernaferrari** [comment]
From Gradle documentation:

The compile configuration still exists but should not be used as it will not offer the guarantees that the api and implementationconfigurations provide.

So, ok, I'll add it.

**Comment by bernaferrari** [comment]
Please test and let me know if it works!

**Comment by mystilleef** [comment]
So far so good. I appreciate that it changes **compile** to **implementation**. 

Maybe I should open another bug report for this, but the only problem I'm seeing that named parameters in Groovy functions and top-level variables are not properly converted to Kotlin. 

For example,

testImplementation(group: "junit", name: "junit", version: "4.12")

This is not valid Kotlin.

Instead, that should be converted to.

testImplementation(group = "junit", name = "junit", version = "4.12")

In other words, ":" should be converted to "="

I know that's not a common style of writing dependencies, but I just thought I should point that out.

Also, I don't believe top-level Groovy variables like **group** and **version** are properly converted to Kotlin. For example "group blah" should be "group = blah". 

That's what I've found so far. I'll do some more testing.

**Comment by bernaferrari** [comment]
Please check: https://github.com/bernaferrari/GradleKotlinConverter/commit/1c445cdddd46ab338fffb1218ad1458f4a0f5eea

It should work fine.
